libs/blockchain-watcher/feature-app/src/lib/store/envStore.ts

```diff
@@ -20,44 +22,69 @@ class EnvStore {
   }
 
   get infuraId(): string {
-    return ''
+    return process.env['INFURA_ID'] || ''
   }
 
-  get ethChainUrl(): string {
+  get ethNetworkUrl(): string {
+    if (this.isTest) {
+      return process.env['ETH_NETWORK_URL_TEST'] || 'http://localhost:8545'
+    }
+
     if (this.isDev) {
-      return 'http://localhost:8545'
+      return process.env['ETH_NETWORK_URL_DEV'] || 'http://localhost:8545'
     }
 
     if (this.isStage) {
-      return `https://ropsten.infura.io/v3/${this.infuraId}`
+      return (
+        process.env['ETH_NETWORK_URL_STAGE'] ||
+        `https://ropsten.infura.io/v3/${this.infuraId}`
+      )
     }
 
     if (this.isProd) {
-      return `https://infura.io/v3/${this.infuraId}`
+      return (
+        process.env['ETH_NETWORK_URL_PROD'] || `https://infura.io/v3/${this.infuraId}`
+      )
     }
 
     return ''
   }
 
   get contractAddress(): string {
-    return process.env.FACTORY_ADDRESS || ''
+    if (this.isTest) {
+      return stringHelper.generateRandomEthAddr()
```

Comment:
`generateRandomEthAddr` can probably be done in a similar way as I proposed for the private key above, e.g. `ethers.Wallet.createRandom().address`